### PR TITLE
[QPG6100] Addition OpenThread support

### LINKED ISSUE #2939: [QPG6100] Enable OpenThread support
#### Problem
QPG6100 first platform port does not support OpenThread yet.

PR with first port:
https://github.com/project-chip/connectedhomeip/pull/2669

 #### Proposed Solution
Add integration for QPG6100 OpenThread in build flow and code

### PR DESCRIPTION
**Pull request by tima-q**
Note - Currently failing on EFR32 tests  - #4527 tracking required update of EFR32 SDK

* Addition OpenThread to QPG6100 platform
* Update relevant QPG6100 build files
* Enabling OT related code in QPG6100 lock-app

* Update openthread submodule to version supporting QPG6100/QPG7015M
** Update GenericThreadstackManagerImpl due to OT API update
** Update to new EFR32 directory structure within OT
** Update Cirque test output folder - was updated by OT to 'simulation'

Fixes #2939

### CODE REVIEW

**Comment by woody-apple** [comment]
@tima-q Sadly still failing on EFR32 :(

**Comment by woody-apple** [comment]
@tima-q - there are conflicts here :(

**Comment by tima-q** [comment]
By the updates done for #4527, the Silabs builds can cope with the upgrade of the OpenThread repo in this PR.
@bzbarsky-apple @woody-apple

**Review by msandstedt — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: examples/platform/qpg6100/app/Service.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+{
+    chip::Inet::IPAddress addr;
+    if (!ConnectivityMgrImpl().IsThreadAttached())
+    {
+        return;
+    }
+    ThreadStackMgrImpl().LockThreadStack();
+    otError error = OT_ERROR_NONE;
+    otMessageInfo messageInfo;
+    otUdpSocket mSocket;
```
Comment (bzbarsky-apple) [inline examples/platform/qpg6100/app/Service.cpp:79]: It's a bit weird to name this `mSocket` if it's not a member...

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/platform/qpg6100/app/Service.cpp (hunk 1 above)
Comment (tima-q) [reply examples/platform/qpg6100/app/Service.cpp:79]: correct - have to admit this is inherited code - that could better be moved to a common location

**Pull request merged**